content/blog/en/version-6-dojo.md

```diff
@@ -0,0 +1,295 @@
+        }}
+        classes={[css.button, css.decrement]}
+      />
+      <div classes={[css.counter]}>{`${count}`}</div>
+      <button
+        onclick={() => {
+          icache.set("count", count + incrementStep);
+        }}
+        classes={[css.button, css.increment]}
+      />
+    </div>
+  );
+});
+```
+
+For more information please see the [Dojo middleware reference guides](/learn/middleware/introduction) or some of the middleware examples on CodeSandbox:
+
+ * [Dojo `theme` middleware](https://codesandbox.io/s/theme-middleware-4btv7)
+ * [Dojo `icache` middleware](https://codesandbox.io/s/advanced-icache-middleware-teeig)
+
+Don't worry, the existing class-based widget APIs are not going away! These enhancements are additive and backwards-compatible, providing what we believe to be an ergonomic improvement on top of the existing widget APIs.
+
+We are really looking forward to seeing the fun and innovative middlewares from the Dojo community, and, as always, please let us know any feedback that you might have!
+
+## Config free Custom Elements
+
+Including Web Components as a first class citizen in Dojo is something that we’ve always been passionate about and now compiling your Dojo widgets to Custom Elements is _even_ easier. In Dojo 6 there is no configuration required, other than defining the widget(s) in the `.dojorc` to instruct the `@dojo/cli-build-widget` to compile your widgets to custom elements. We think this is a significant improvement to current tooling that requires widgets to be explicitly configured with the custom element details. The approach prior to Dojo version 6 required additional development effort, foresight, and created an additional maintenance burden of keeping the configuration up to date with changes to widget properties. This can be seen below with the custom element configuration for an example widget.
+
+Configuration required prior to version 6 for compiling a Dojo widget to a custom element:
+
+> .dojorc
+```json
+{
+  "build-widget": {
+    "widgets": [
+      "src/MyWidget"
+    ]
+  }
+}
+```
+
+> src/MyWidget
+```tsx
+import { WidgetBase } from '@dojo/framework/core/WidgetBase';
+import { customElement } from '@dojo/framework/core/decorators/customElement';
+import { CustomElementChildType } from '@dojo/framework/core/registerCustomElement';
+
+interface MyWidgetProperties {
+  value: string;
+  disabled: boolean;
+  onInput: (input: value) => void;
+}
+
+@customElement<MyWidgetProperties>({
+	tag: 'dojo-my-widget',
+	childType: CustomElementChildType.TEXT,
+	properties: ['disabled'],
+	attributes: ['value'],
+	events: [
+		'onInput'
+	]
+})
+class MyWidget extends WidgetBase<MyWidgetProperties> {
+  protected render() {
+    // rendering
+  }
+}
+```
+
+In Dojo version 6, this is substantially simplified. Now the Dojo build tool's intelligent configuration of custom elements only requires the `.dojorc` entry and will automatically include new, changed or removed properties!
```

Comment:
I love this old v new pattern that is used throughout this post. It not only highlights what changed but gives a clear direction on how to adopt those changes.